packages/stakers/src/set/ensureStakerPkgsNetworkConfig.ts

```diff
@@ -0,0 +1,194 @@
+  delete compose.compose.networks?.[params.DOCKER_STAKER_NETWORK_NAME];
+  compose.write();
+  // remove from compose service network
+  for (const [serviceName, service] of Object.entries(
+    compose.compose.services
+  )) {
+    const composeService = new ComposeFileEditor(dnpName, false);
+    // network declared in array format is not supported
+    if (Array.isArray(service.networks)) {
+      logs.warn(
+        `Service ${serviceName} in ${dnpName} has a network declared in array format, skipping`
+      );
+      continue;
+    }
+    composeService
+      .services()
+      // eslint-disable-next-line no-unexpected-multiline
+      [serviceName].removeNetwork(params.DOCKER_STAKER_NETWORK_NAME);
+    composeService.write();
+  }
+}
+
+async function disconnectConnectedPkgFromStakerNetwork(
+  networkName: string,
+  pkgContainers: PackageContainer[]
+): Promise<void> {
+  const connectedContainers = pkgContainers.filter((container) =>
+    container.networks.some((network) => network.name === networkName)
+  );
+  for (const container of connectedContainers)
+    await dockerNetworkDisconnect(networkName, container.containerName);
+}
+
+/**
+ * Adds the staker network and its fullnode alias to the docker-compose file
+ */
+function addStakerNetworkToCompose(
+  dnpName: string,
+  alias?: string | null
+): void {
+  // add to compose network
+  const compose = new ComposeFileEditor(dnpName, false);
+  const stakerNetwork =
+    compose.compose.networks?.[params.DOCKER_STAKER_NETWORK_NAME];
+  if (!stakerNetwork) {
+    compose.compose.networks = {
+      ...compose.compose.networks,
+      [params.DOCKER_STAKER_NETWORK_NAME]: {
+        external: true,
+      },
+    };
+    compose.write();
+  }
+  // add to compose service network
+  for (const [serviceName, service] of Object.entries(
+    compose.compose.services
+  )) {
+    const composeService = new ComposeFileEditor(dnpName, false);
+    // network declared in array format is not supported
+    if (Array.isArray(service.networks)) {
+      logs.warn(
+        `Service ${serviceName} in ${dnpName} has a network declared in array format, skipping`
+      );
+      continue;
+    }
+    const stakerServiceNetwork =
+      service.networks?.[params.DOCKER_STAKER_NETWORK_NAME];
+
+    if (!stakerServiceNetwork) {
+      composeService
+        .services()
+        // eslint-disable-next-line no-unexpected-multiline
+        [serviceName].addNetwork(params.DOCKER_STAKER_NETWORK_NAME, {
+          aliases: alias ? [alias] : [],
+        });
+      composeService.write();
+    }
+  }
+}
+
+/**
+ * Connects the staker pkg to the staker network with the fullnode alias
+ */
+async function connectPkgToStakerNetwork(
+  networkName: string,
+  pkgContainers: PackageContainer[],
+  alias?: string | null
+): Promise<void> {
+  for (const container of pkgContainers)
+    await dockerNetworkConnectNotThrow(networkName, container.containerName, {
```

Comment:
same as above, why use notThrow and no error handling? being successfully connected tot his new staker network seems important enough to check it